### PR TITLE
Add Terraform testing in Github Action

### PR DESCRIPTION
**Pull request by kobrikx**
### ***List of changes:***

   Added Terraform code automatic testing with TFLint+ReviewDog with Checkov aftercheck.
   It uses:
1.   _Terraform init_ for terraform code initial testing for syntax errors. It also needs as
       _TFLint+ReviewDog_ won't show you anything until you correct syntax errors. 
2.   _TFLint+ReviewDog_. It provides logic testing and show it on PR or Pushes with Stop and 
     show errors in code diff.
3.   _Checkov_ aftercheck required for independent testing. It doesn't shows in code like 
     ReviewDog, but you can see it's logs in CLI.Also _Checkov_ doesn't generate Stop.
4. For Github integration and to display menus with errors on PR it uses _ReviewDog Github App_
You need to install it to your github account: https://github.com/apps/reviewdog

#### ***On push:***

 _ReviewDog:_
ReviewDog on push

 _Checkov:_
Checkov on push

_Pull_request check skips on push testing:_
ReviewDog on push_PR_check_skipped

#### ***On pull_request:***

 _Check window:_
PR_check

 _Diff on PR:_
diff on PR

 _ReviewDog:_
on PR

 _Checkov on PR:_
PR_checkov

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,20 +1,71 @@
...
     runs-on: ubuntu-latest
     steps:
-      - name: Checkout Code
-        uses: actions/checkout@v2-beta
+      - name: Checkout
+        uses: actions/checkout@v2
 
-      - name: Create AWS Profile
-        run: make test
+      - name: Terraform init
```
Comment (DimmKirr) [inline .github/workflows/main.yml:41]: Let's rename this to reflect actual name of the step

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -1,20 +1,71 @@
...
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -reporter=github-pr-check -name=github-pr-check-info
+          
+      - name: Run reviewdog (github-pr-check with fail-on-error)  
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -name=pr-check-fail-on-error -reporter=github-pr-check -fail-on-error -filter-mode=nofilter
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: Let's use common naming for containers (lowercase)

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -1,20 +1,71 @@
...
+  Checkov-check:
+    name: Checkov (github-check)
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+         
+      - name: Run Checkov CLI test
+        continue-on-error: true
+        run: docker run --rm --name Checkov -v $GITHUB_WORKSPACE:/tf bridgecrew/checkov -d /tf -o cli
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: Let's use common naming for container names. Let's not use $GITHUB_WORKSPACE, but instead use $(pwd) or $PWD

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -1,20 +1,71 @@
...
+      - name: Setup Reviewdog
+        run: |
+         mkdir -p $HOME/bin && curl -sfL https://raw.githubusercontent.com/reviewdog/reviewdog/master/install.sh| sh -s -- -b $HOME/bin
+         echo ::add-path::$HOME/bin
+         
+      - name: Run ReviewDog (github-check)
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -name="github-check" -reporter=github-check -level=warning -filter-mode=nofilter
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: Let's use `$(pwd)` or `$PWD` instead of `$GITHUB_WORKSPACE`

File: .github/workflows/main.yml [hunk 5]
```diff
@@ -1,20 +1,71 @@
...
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -reporter=github-pr-check -name=github-pr-check-info
+          
+      - name: Run reviewdog (github-pr-check with fail-on-error)  
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -name=pr-check-fail-on-error -reporter=github-pr-check -fail-on-error -filter-mode=nofilter 
+
+  Checkov-check:
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: Let's use lowercase naming

File: .github/workflows/main.yml [hunk 6]
```diff
@@ -1,20 +1,71 @@
...
+         
+      - name: Run ReviewDog (github-check)
+        env:
+          REVIEWDOG_GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          docker run --rm --name TFLint -v $GITHUB_WORKSPACE:/data -t wata727/tflint --format=checkstyle | reviewdog -f=checkstyle -name="github-check" -reporter=github-check -level=warning -filter-mode=nofilter
+          
+  reviewdog-pr:
+    if: github.event_name == 'pull_request'
+    name: reviewdog on Pull Request
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: Capitalize to `ReviewDog`

File: .github/workflows/main.yml [hunk 7]
```diff
@@ -1,20 +1,71 @@
...
-        uses: actions/checkout@v2-beta
+      - name: Checkout
+        uses: actions/checkout@v2
 
-      - name: Create AWS Profile
-        run: make test
+      - name: Terraform init
+        run: terraform init
+            
+      - name: Setup reviewdog
```
Comment (DimmKirr) [inline .github/workflows/main.yml]: `ReviewDog`

File: .github/workflows/main.yml (hunk 2 above)
Comment (DimmKirr) [inline .github/workflows/main.yml]: `$PWD` or `$(pwd)`

**Review by DimmKirr — CHANGES_REQUESTED** [review]

File: .github/workflows/main.yml (hunk 3 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml (hunk 4 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml (hunk 5 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done

File: .github/workflows/main.yml (hunk 2 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml (hunk 7 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml (hunk 6 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

File: .github/workflows/main.yml (hunk 2 above)
Comment (kobrikx) [reply .github/workflows/main.yml]: Done.

**Pull request merged**